ctru-rs/src/services/cam.rs

```diff
@@ -21,191 +20,165 @@ pub struct Cam {
-        const LETTER    = ctru_sys::PHOTO_MODE_LETTER;
-    }
+/// Flag to pass to [Camera::set_photo_mode]
+#[derive(Copy, Clone, Debug)]
+#[repr(u32)]
+pub enum PhotoMode {
+    Normal = ctru_sys::PHOTO_MODE_NORMAL,
+    Portrait = ctru_sys::PHOTO_MODE_PORTRAIT,
+    Landscape = ctru_sys::PHOTO_MODE_LANDSCAPE,
+    Nightview = ctru_sys::PHOTO_MODE_NIGHTVIEW,
+    Letter = ctru_sys::PHOTO_MODE_LETTER,
 }
 
-bitflags! {
-    /// A set of flags to be passed to [Camera::set_effect]
-    #[derive(Default)]
-    pub struct CamEffect: u32 {
-        const NONE     = ctru_sys::EFFECT_NONE;
-        const MONO     = ctru_sys::EFFECT_MONO;
-        const SEPIA    = ctru_sys::EFFECT_SEPIA;
-        const NEGATIVE = ctru_sys::EFFECT_NEGATIVE;
-        const NEGAFILM = ctru_sys::EFFECT_NEGAFILM;
-        const SEPIA01  = ctru_sys::EFFECT_SEPIA01;
-    }
+/// Flag to pass to [Camera::set_effect]
+#[derive(Copy, Clone, Debug)]
+#[repr(u32)]
+pub enum Effect {
+    None = ctru_sys::EFFECT_NONE,
+    Mono = ctru_sys::EFFECT_MONO,
+    Sepia = ctru_sys::EFFECT_SEPIA,
+    Negative = ctru_sys::EFFECT_NEGATIVE,
+    Negafilm = ctru_sys::EFFECT_NEGAFILM,
+    Sepia01 = ctru_sys::EFFECT_SEPIA01,
 }
 
-bitflags! {
-    /// A set of flags to be passed to [Camera::set_contrast]
-    #[derive(Default)]
-    pub struct CamContrast: u32 {
-        const PATTERN_01 = ctru_sys::CONTRAST_PATTERN_01;
-        const PATTERN_02 = ctru_sys::CONTRAST_PATTERN_02;
-        const PATTERN_03 = ctru_sys::CONTRAST_PATTERN_03;
-        const PATTERN_04 = ctru_sys::CONTRAST_PATTERN_04;
-        const PATTERN_05 = ctru_sys::CONTRAST_PATTERN_05;
-        const PATTERN_06 = ctru_sys::CONTRAST_PATTERN_06;
-        const PATTERN_07 = ctru_sys::CONTRAST_PATTERN_07;
-        const PATTERN_08 = ctru_sys::CONTRAST_PATTERN_08;
-        const PATTERN_09 = ctru_sys::CONTRAST_PATTERN_09;
-        const PATTERN_10 = ctru_sys::CONTRAST_PATTERN_10;
-        const PATTERN_11 = ctru_sys::CONTRAST_PATTERN_11;
-
-        const LOW    = ctru_sys::CONTRAST_LOW;
-        const NORMAL = ctru_sys::CONTRAST_NORMAL;
-        const HIGH   = ctru_sys::CONTRAST_HIGH;
-    }
+/// Flag to pass to [Camera::set_contrast]
+#[derive(Copy, Clone, Debug)]
+#[repr(u32)]
+pub enum Contrast {
+    Low = ctru_sys::CONTRAST_LOW,
+    Normal = ctru_sys::CONTRAST_NORMAL,
+    High = ctru_sys::CONTRAST_HIGH,
 }
 
-bitflags! {
-    /// A set of flags to be passed to [Camera::set_lens_correction]
-    #[derive(Default)]
-    pub struct CamLensCorrection: u32 {
-        const OFF   = ctru_sys::LENS_CORRECTION_OFF;
-        const ON_70 = ctru_sys::LENS_CORRECTION_ON_70;
-        const ON_90 = ctru_sys::LENS_CORRECTION_ON_90;
+/// Flag to pass to [Camera::set_lens_correction]
+#[derive(Copy, Clone, Debug)]
+#[repr(u32)]
+pub enum LensCorrection {
+    Off = ctru_sys::LENS_CORRECTION_DARK,
+    Normal = ctru_sys::LENS_CORRECTION_NORMAL,
+    Bright = ctru_sys::LENS_CORRECTION_BRIGHT,
```

Comment:
I assume these are synonyms for `OFF`, `70`, `90` ? Maybe comments indicating that would be helpful